src/filter.rs

```diff
@@ -0,0 +1,300 @@
+use libc::{c_int, c_char, c_void, size_t};
+use std::{mem, slice, ptr};
+use {raw, call, util, Error};
+use std::ffi::{CString, CStr};
+use std::path::PathBuf;
+use raw::{git_filter, git_writestream};
+
+
+#[repr(C)]
+struct Filter<T> {
+    filter: git_filter,
+    instance: T,
+}
+
+#[repr(C)]
+struct Stream<T> {
```

Comment:
Perhaps this could be named `RawWriteStream` and the above type could be named `RawFilter`?